### PR TITLE
MiddlewareException

### PR DESCRIPTION
**Pull request by saraialvarezrosas**

### CODE REVIEW

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/MiddlewareException.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.jalau.at18.searchobject.common.exception;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/MiddlewareException.java:9]: Please add documentation

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java [hunk 2]
```diff
@@ -7,14 +7,21 @@
...
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.io.FileWriter;
 import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Scanner;
 import java.util.logging.Logger;
 import javax.servlet.*;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java]: Please use single imports

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java [hunk 3]
```diff
@@ -31,34 +38,72 @@ public class EmotionRecognitionMiddleware implements Filter {
...
+                    //Verify that a field isn't empty
+                    if(!req.getParameter("token").isEmpty()) {
+                        LOG.info("EMOTION RECOGNITION ---- RUNNING -------");
+                        chain.doFilter(request, response);
+                    } else {
+                        LOG.warning(" ENTER AN TOKEN ");
+                        throw new MiddlewareException("    Enter an token ");
+                    }
+                }  else {
+                        LOG.warning(" ENTER AN IMAGE ");
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java:86]: Please correct the indentation

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java [hunk 4]
```diff
@@ -31,34 +38,72 @@ public class EmotionRecognitionMiddleware implements Filter {
...
+                pw.flush(); //delete the content of the txt file.
+                pw.close();
+                fw.close();
+                throw new MiddlewareException("    Token has no more uses, please request another one");
+            } else {
+                LOG.warning(" GENERATED TOKEN");
+                throw new MiddlewareException("    Generated token ");
+            }
+        } catch (InstantiationError | MiddlewareException e) {
+                    LOG.warning(" ERROR LOADING THE MODEL " + e);
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java]: Please correct the indentation

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java [hunk 5]
```diff
@@ -7,14 +7,21 @@
  * Information and shall use it only in accordance with the terms of the
  * Licence agreement you entered into with Jalasoft
  */
+import org.jalau.at18.searchobject.common.exception.MiddlewareException;
 import org.jalau.at18.searchobject.common.logger.At18Logger;
 import javax.servlet.*;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java]: Please use single imports

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java [hunk 6]
```diff
@@ -29,35 +36,72 @@ public class FaceDetectionControllerMiddleware implements Filter {
...
+                    //Verify that a field isn't empty
+                    if(!req.getParameter("type").isEmpty()) {
+                        LOG.info("FACE DETECTION ---- RUNNING -------");
+                        chain.doFilter(request, response);
+                    } else {
+                        LOG.warning(" THE TYPE FIELD IS EMPTY ");
+                        throw new MiddlewareException("    The type field is empty ");
+                    }
+                }  else {
+                        LOG.warning(" ENTER AN IMAGE ");
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java:84]: Please correct the indentation

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java [hunk 7]
```diff
@@ -29,35 +36,72 @@ public class FaceDetectionControllerMiddleware implements Filter {
...
             } else {
-                LOG.warning(" ENTER AN IMAGE ");
+                LOG.warning(" GENERATED TOKEN ");
+                throw new MiddlewareException("    Generated token ");
             }
-        } catch (InstantiationError e) {
-            LOG.warning(" ERROR LOADING THE MODEL "+e);
-            e.printStackTrace();
+        } catch (InstantiationError | MiddlewareException e) {
+                LOG.warning(" ERROR LOADING THE MODEL " + e);
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java]: Please correct the indentation

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java [hunk 8]
```diff
@@ -7,14 +7,21 @@
...
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.io.FileWriter;
 import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Scanner;
 import java.util.logging.Logger;
 import javax.servlet.*;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java]: Please use single imports

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/token/GenerateToken.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
+package org.jalau.at18.searchobject.middleware.token;
+
+import org.jalau.at18.searchobject.common.logger.At18Logger;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import java.io.File;
+import java.io.FileWriter;
+        import java.io.IOException;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/token/GenerateToken.java]: Please correct the indentation

**Review by rodri-98 — CHANGES_REQUESTED** [review]

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java [hunk 10]
```diff
@@ -29,35 +36,72 @@ public class FaceDetectionControllerMiddleware implements Filter {
...
+                    //Verify that an empty or null file isn't entered
+                if(req.getPart("file").getSize() != 0L && req.getPart("file").getSize() > 100 && req.getPart("file").getContentType() != null) {
+                    LOG.info(" ACCEPT THE IMAGE ");
+                    //Verify that a field isn't empty
+                    if(!req.getParameter("type").isEmpty()) {
+                        LOG.info("FACE DETECTION ---- RUNNING -------");
+                        chain.doFilter(request, response);
+                    } else {
+                        LOG.warning(" THE TYPE FIELD IS EMPTY ");
+                        throw new MiddlewareException("    The type field is empty ");
```
Comment (DaniMB95) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java:81]: the status error should not also appear here?

Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java:81]: no, the status error is sent in the catch

**Review by rodri-98 — APPROVED** [review]
Good job! , before to merge please solve the conflicts

**Comment by DaniMB95** [comment]
Solve the conflict before merge your branch

**Pull request merged**